sources/Valkey.Glide/Internals/Request.StreamCommands.cs

```
@@ -0,0 +1,868 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+using static Valkey.Glide.Internals.FFI;
+
+namespace Valkey.Glide.Internals;
+
+internal partial class Request
+{
+    public static Cmd<GlideString, ValkeyValue> StreamAddAsync(ValkeyKey key, ValkeyValue messageId, long? maxLength, ValkeyValue minId, bool useApproximateTrimming, NameValueEntry[] streamPairs, long? limit, bool noMakeStream)
+    {
+        List<GlideString> args = [key.ToGlideString()];
+
+        // Add NOMKSTREAM if specified
+        if (noMakeStream)
+        {
+            args.Add("NOMKSTREAM".ToGlideString());
+        }
+
+        // Add trimming options (MAXLEN or MINID)
+        if (maxLength.HasValue)
```

Are we actually enforcing the fact that `maxLength` and `minId`? It seems like, if both were provided under the current implementation, we would just silently continue with `maxLength`?

---

Will update with proper enforcement.

---

Great, thank you!